### PR TITLE
Workflow improvements

### PR DESCRIPTION
**Pull request by kostko**
* Add missing workflow input support.
* Add `data_by_slug` filter.

### CODE REVIEW

File: resolwe/flow/tests/test_expression_filters.py [hunk 1]
```diff
@@ -60,6 +61,7 @@ def test_templatetags(self):
 re-save basename "{{ '/foo/bar/moo' | basename }}"
 re-save subtype "{{ 'data:test:inputobject:' | subtype('data:') }}"
 re-save yesno "{{ true | yesno('yes', 'no') }}"
+re-save datalookup "{{ 'input-data-object' | lookup_data_by_slug }}"
```
Comment (mstajdohar) [inline resolwe/flow/tests/test_expression_filters.py]: How about just `by_slug`?

Comment (mstajdohar) [inline resolwe/flow/tests/test_expression_filters.py]: > But we can remove the `lookup_` prefix.

Comment (kostko) [inline resolwe/flow/tests/test_expression_filters.py]: Done.

**Comment by dblenkus** [comment]
@mstajdohar Where do we need this? 
We should be careful with such features, because slug is not always exactly what you set - suffix can be added to it to make it unique.

**Comment by kostko** [comment]
> @mstajdohar Where do we need this?

@jkokosar needs this when constructing workflows to reference existing data objects. I think this was also part of the specification.

**Comment by mstajdohar** [comment]
One example is for automating the primary analysis. Janez will prepare certain data objects on the platform (genome, gene model, ontology...) and then in the configuration specify them as inputs to the pipeline.

Another example is that we wish to make the GO Enrichment Analysis process easy to use. Based on the inputs it can figure out which annotation files it should use (human or mouse). The process would expect either human-gaf or mouse-gaf data object on the platform. If they would not be available, it would fail.

Specifically, when you run the GO Enrichment Analysis all that is really needed on input are gene IDs. We can then figure out automatically if the IDs are mouse or human. Based on this we can find the appropriate GAF on the platform automatically, so there is no need for the user to specify it.

**Comment by kostko** [comment]
Can we merge this?

**Comment by mstajdohar** [comment]
I agree. @dblenkus?

**Comment by kostko** [comment]
I've now renamed `inputs` to `input`.

**Pull request merged**